hibernate-core/src/main/java/org/hibernate/loader/BatchLoadSizingStrategy.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.loader;
+
+/**
+ * Strategy (pluggable) for determining an optimal size for batch loads.
+ *
+ * @author Steve Ebersole
+ */
+public interface BatchLoadSizingStrategy {
+	int determineOptimalBatchLoadSize(int numberOfKeyColumns, int numberOfKeys);
```

Comment:
Ah, looking further it seems dialect authors are the intended implementors of this contract, not users as I first thought.